Schemas.md

```diff
@@ -0,0 +1,528 @@
+
+```python
+{
+    'GET': {
+        'apply_draft': {
+            'schema': {
+                'type': 'bool'
+             },
+            'default_value': False
+        }
+    }
+}
+```
+
+**Example when default value is not provided**: Suppose `make_community_owned` is an optional argument which should not take any default value if no value for that argument is provided with the request. In that case, the schema for `make_community_owned` should look like:
+
+```python
+{
+    'PUT':{
+        'make_community_owned': {
+            'schema': {
+                'type': 'bool'
+             },
+            'default_value': None
+        }
+    }
+}
+```
+
+For more information [refer to our examples](#examples-for-reference).
+
+##### Domain object arguments: Schemas with type `object_dict`
+
+Objects which are represented by classes written in the domain layer of the codebase are called domain objects. These classes typically include methods to validate their objects.
+
+For validating domain objects through the SVS, there are two preferred solutions:
+
+###### Case 1: `object_class` key
+
+Many of the domain objects do not get initialized with dictionaries,so they must be initialized by using their `from_dict()` methods.  This case applies when the data coming from a request is a dict and the domain object the dict represents has a `from_dict()` method.
+
+In this case, schemas should have the following two keys:
+
+1. **type**: 'object_dict'
+2. **object_class**: class written in the domain layer of the codebase for the corresponding argument.
+
+**Example**: Let `new_rules` be the list of dicts where each dict item is a representation of the `PlatformParameterRule` domain object in the `platform_parameter_domain.py` file. The schema for `new_rules` should look like:
+
+```python
+HANDLER_ARGS_SCHEMAS = {
+    'POST': {
+        'new_rules': {
+            'schema': {
+                'type': 'list',
+                'items': {
+                    'type': 'object_dict',
+                    'object_class': (
+                        platform_parameter_domain.PlatformParameterRule)
+                }
+            }
+        }
+    }
+}
+```
+
+###### Case 2: `validation_method` key
+
+Sometimes, a domain object has a `validate_dict()` method that should be used for validation. These methods sometimes need extra arguments, for example to enforce strict validation. Since these cases cannot be handled generally, you'll need to write a function that accepts the dictionary from the request and passes it to the domain object's `validate_dict()` method, specifying other parameters as needed. Your function should be in the `domain_objects_validator.py` file.
+
+The newly written function of the `domain_objects_validator.py` file should be directly passed into the schema with a schema key named `validation_method`. Schemas for these cases should have the two keys as follows:
```

Comment:
two keys as follows --> two keys set up as follows

(or similar, this might need some thought)